components/supervisor/pkg/terminal/terminal.go

```diff
@@ -143,6 +143,10 @@ func (m *Mux) doClose(ctx context.Context, alias string) error {
 		log.WithError(err).Warn("Error while closing pseudo-terminal")
 	}
 
+	if forceSuccess {
```

Comment:
Should this line not be called before `term.Close` to avoid possible races? :thinking: